cfripper/rules/s3_public_access.py

```diff
@@ -105,3 +105,44 @@ def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
                     context={"config": self._config, "extras": extras, "logical_id": logical_id, "resource": resource},
                 )
         return result
+
+
+class S3BucketPublicReadAclRule(Rule):
+    """
+    Checks if any S3 bucket policy has access control set to `PublicRead`.
+
+    Risk:
+        Unless the bucket is hosting static content, S3 buckets should not have Public Read available on a bucket.
+        This allows anyone to read any objects to your S3 bucket.
+
+    Fix:
+        Remove any configuration that looks like `"AccessControl": "PublicRead"` from your S3 bucket policy.
+
+    Filters context:
+        | Parameter     | Type               | Description                                                    |
+        |:-------------:|:------------------:|:--------------------------------------------------------------:|
+        |`config`       | str                | `config` variable available inside the rule                    |
+        |`extras`       | str                | `extras` variable available inside the rule                    |
+        |`logical_id`   | str                | ID used in Cloudformation to refer the resource being analysed |
+        |`resource`     | `Resource`         | S3 Bucket that is being addressed                              |
+    """
+
+    GRANULARITY = RuleGranularity.RESOURCE
+    REASON = "S3 Bucket {} should not have a public-read acl"
+    RISK_VALUE = RuleRisk.HIGH
+
+    def invoke(self, cfmodel: CFModel, extras: Optional[Dict] = None) -> Result:
+        result = Result()
+        for logical_id, resource in cfmodel.Resources.items():
+            if (
+                resource.Type == "AWS::S3::Bucket"
+                and hasattr(resource, "Properties")
+                and resource.Properties.get("AccessControl") == "PublicRead"
```

Comment:
You can use the resources_filtered_by_type method to filter resources if you want:
```suggestion
        for logical_id, resource in cfmodel.resources_filtered_by_type(("AWS::S3::Bucket", )).items():
            if (
                hasattr(resource, "Properties")
                and resource.Properties.get("AccessControl") == "PublicRead"
```

Is it possible to have a S3Bucket without Prooperties 🤔 ?

Reply:
I will say no, but not sure about it 100%